internal/conditions/conditions.go

```diff
@@ -122,7 +123,21 @@ func (u *updater) SetConditionsFailed(ctx context.Context, cr *appsv1alpha1.NIMS
 		Message: message,
 	})
 	cr.Status.State = v1alpha1.NIMServiceStatusFailed
-	return u.client.Status().Update(ctx, cr)
+	return u.updateNIMServiceStatus(ctx, cr)
+}
+
+func (u *updater) updateNIMServiceStatus(ctx context.Context, cr *appsv1alpha1.NIMService) error {
+
+	obj := &appsv1alpha1.NIMService{}
+	errGet := u.client.Get(ctx, types.NamespacedName{Name: cr.Name, Namespace: cr.GetNamespace()}, obj)
+	if errGet != nil {
+		return errGet
+	}
+	obj.Status = cr.Status
+	if err := u.client.Status().Update(ctx, obj); err != nil {
```

Comment:
Can we use Patch() operation instead of Get then Update? It should be more efficient

Reply:
Since the status field has list of conditions, I am assuming I will have to use MergePatch. I will need the original object to create the merge patch.

Reply:
agree, patch or server side apply is more resilient for this issue. Also feel free to update this logic across all types as a separate PR.